bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/ApiBridge.java

```diff
@@ -0,0 +1,233 @@
+                    throw new NetatmoException("Required scopes missing to access : " + typeOfRest);
+                }
+                managers.put(typeOfRest, tentative);
+            } catch (SecurityException | ReflectiveOperationException | NetatmoException e) {
+                logger.error("Error invoking RestManager constructor for class {} : {}", typeOfRest, e.getMessage());
+            }
+        }
+        return (T) managers.get(typeOfRest);
+    }
+
+    synchronized <T> T executeUri(URI uri, HttpMethod method, Class<T> classOfT, @Nullable String payload)
+            throws NetatmoException {
+        try {
+            logger.trace("executeUri {}  {} ", method.toString(), uri);
+
+            Request request = httpClient.newRequest(uri).method(method).timeout(TIMEOUT_MS, TimeUnit.MILLISECONDS);
+
+            httpHeaders.entrySet().forEach(entry -> request.header(entry.getKey(), entry.getValue()));
+
+            if (payload != null && (HttpMethod.POST.equals(method) || HttpMethod.PUT.equals(method))) {
+                InputStream stream = new ByteArrayInputStream(payload.getBytes(StandardCharsets.UTF_8));
+                try (InputStreamContentProvider inputStreamContentProvider = new InputStreamContentProvider(stream)) {
+                    request.content(inputStreamContentProvider, null);
+                }
+                if (!NetatmoConstants.URL_API.contains(uri.getHost())) {
+                    request.getHeaders().remove(HttpHeader.CONTENT_TYPE);
+                    request.header(HttpHeader.CONTENT_TYPE, "application/json;charset=utf-8");
+                }
+            }
+
+            ContentResponse response = request.send();
+
+            Code statusCode = HttpStatus.getCode(response.getStatus());
+            String responseBody = new String(response.getContent(), StandardCharsets.UTF_8);
+            logger.trace("executeUri returned : {} ", responseBody);
+
+            if (statusCode == Code.OK) {
+                return deserialize(classOfT, responseBody);
+            } else {
+                NetatmoException exception;
+                if (statusCode == Code.BAD_REQUEST) {
+                    ApiError error = deserialize(ApiError.class, responseBody);
+                    exception = new NetatmoException(error.getCode(), error.getMessage());
+                } else {
+                    exception = new NetatmoException(statusCode.getCode(), responseBody);
+                    prepareReconnection(exception);
+                }
+                throw exception;
+            }
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            throw new NetatmoException(String.format("Exception while calling %s", uri.toString()), e);
+        }
+    }
+
+    public <T> T deserialize(Class<T> classOfT, String serviceAnswer) throws NetatmoException {
+        try {
+            T result = gson.fromJson(serviceAnswer, classOfT);
```

Comment:
Warning:
`ApiBridge.java:[196,24] Null type mismatch (type annotations): required 'T' but this expression has type '@Nullable T', where 'T' is a free type variable`

Reply:
Corrected.